src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -515,7 +515,13 @@ static bool inner_resolve_as_ip_literal_locked(
     *port = default_port;
   }
   grpc_resolved_address addr;
-  *hostport = grpc_core::JoinHostPort(*host, atoi(port->c_str()));
+  int port_number = atoi(port->c_str());
+  if (strcmp(port->c_str(), "http") == 0) {
```

Comment:
@apolcyn Is there a portable way to look up the port name in /etc/services?

The native DNS resolver code also [hard-codes these values](https://github.com/grpc/grpc/blob/61af6e7f4f25804a0140de96cd5a1983ec6e16a7/src/core/lib/iomgr/resolve_address_posix.cc#L88), but it looks like that code only kicks in if `getaddrinfo()` doesn't understand the port name, so I assume it checks /etc/services first.  It would be nice to get the same behavior here if we can.

Reply:
We could probably try out `getservbyname` (I found https://github.com/grpc/grpc/pull/19285 where this was brought up earlier), if we wanted to get rid of the hardcoding entirely.

Note that this PR is a basically an alternative to https://github.com/grpc/grpc/pull/19285 -- #19285 in it's current state centralizes the hardcoded map port number map, as another alternative.

Reply:
Thanks @apolcyn! Would you mind reopening that #19285? It looks much more comprehensive and I'm fine with discarding this pr.

Reply:
Sure, I've just updated and reopened https://github.com/grpc/grpc/pull/19285